daemon/algod/api/algod.oas3.yml

```diff
@@ -2023,6 +2035,77 @@
         },
         "type": "object"
       },
+      "SimulationOpcodeTraceUnit": {
+        "description": "The set of trace information and effect from evaluating a single opcode.",
+        "properties": {
+          "pc": {
+            "description": "The program counter of the current opcode being evaluated.",
+            "type": "integer"
+          }
+        },
+        "required": [
+          "pc"
+        ],
+        "type": "object"
+      },
+      "SimulationPcToInnerIndex": {
+        "description": "The mapping from a program counter to an index into inner app call traces, indicating which step contract simulation branches off into an inner txn.",
+        "properties": {
+          "inner-index": {
+            "description": "The index into inner trace array, to which inner transaction get branched off from program execution.",
+            "type": "integer"
+          },
+          "pc": {
+            "description": "The program counter of the current opcode being evaluated.",
+            "type": "integer"
+          }
+        },
+        "required": [
+          "inner-index",
+          "pc"
+        ],
+        "type": "object"
+      },
+      "SimulationTransactionExecTrace": {
+        "description": "The execution trace of calling an app or a logic sig, containing the inner app call trace in a recursive way.",
+        "properties": {
+          "inner-trace": {
+            "description": "An array of SimulationTransactionExecTrace representing calling inner transactions to apps.",
+            "items": {
+              "$ref": "#/components/schemas/SimulationTransactionExecTrace"
+            },
+            "type": "array"
+          },
+          "step-to-inner-map": {
+            "description": "The map from the step index of trace to the inner trace index, indicating which step contract simulation branches off into an inner txn.",
```

Comment:
Does this step index of the trace always point to a `itxn_submit` opcode?  that might be a useful comment to include.
also I can see maybe that we want to do `--allow-more-inners`

Reply:
As of current implementation, the step index of trace doesn't always point to `itxn_submit`, for I only record inner app call trace, but we can discuss and change the implementation.

The consistency looks interesting, if we take your suggestion, then I think we should always return trace for all kinds of transaction, not just app calls?

Reply:
after a few thoughts, I think it would be interesting, and it is a good reminder that `itxn_submit` may lead to multiple inner txn, so I fixed the testcase and add comments in oas2 model: [oas2 trace step to inner map comment, change internal impl from map t…](https://github.com/algorand/go-algorand/pull/5360/commits/7f7e18126432db1b0a2c7dfa28c8695df77d62e3)